core/ts.core/src/ts/resources/TypeScriptProject.java

```diff
@@ -134,13 +150,20 @@ public final ITypeScriptServiceClient getClient() throws TypeScriptException {
 					this.client = createServiceClient(getProjectDir());
 					copyListeners();
 					onCreateClient(client);
+					// determine root files and project name
+					String projectName = projectDir.getCanonicalPath();
+					List<String> rootFiles = new ArrayList<>();
+					for (String tsconfigFilePath : getTsconfigFilePaths()) {
+						rootFiles.add(new File(projectDir, tsconfigFilePath).getCanonicalPath());

```

Comment:
Please use FileUtils.getPath(new File(projectDir, tsconfigFilePath)) to normalize path.